spec/models/user_spec.rb

```diff
@@ -188,4 +188,24 @@
       end
     end
   end
+
+  context "when updating name" do
+    it "is valid to leave name unset" do
+      user = create :user, :with_no_name
+      user.name = nil
+      expect(user.valid?).to be true
```

Comment:
```suggestion
      user = build :user, :with_no_name
      expect(user).to be_valid
```

I switched the `create` to `build` because every time you use `create` it creates an actual entry in the DB which is not always necessary to check if a class instance is valid. For this reason `build` and `build_stub` are performant than `create`. 

Also could we not use `let` statements at the top of the specs to create instances and pass in different contexts as we test them? For example https://github.com/alphagov/forms-admin/blob/main/spec/models/draft_question_spec.rb#L37-L56 you will see we testing `question_text` but the instance of DraftQuestion is only defined once at the top of the spec file. Sometimes you do need a a different instance but I dont think we need to repeat creating instances for each specification

Reply:
https://github.com/thoughtbot/factory_bot/blob/main/GETTING_STARTED.md#build-strategies Not one for now but we should look at our build strategies that we are using. I would try to use `build` more than `create` because it doesn't "touch" the database like "create" does but also sometimes `build_stubbed` might be a better option for testing and would be more performant. what we dont want is for our test suite to grind to a halt because even test is constantly reading/writing to the database. I've been in teams where test suites have taken like 30mins to run and by switching build strategies for tests helped almost half the time tests took to run.

Reply:
Unfortunately with this test group `build` doesn't always work, because the validation we're testing requires a persistence layer:

https://github.com/alphagov/forms-admin/blob/ce694cfae204154cbfe09008244a38b54a7a994f/app/models/user.rb#L69-L71

Using `build` instead of `create` causes the tests as written to fail.

Reply:
oh I see, what about the suggestion of using lets?

Reply:
I had a look at using `let`, I think for this set of tests we'd need a different context for each test... I'm not sure if that's better than doing the `create` in the test, what do you think?

Reply:
Happy to just keep it as is for the time being.  We can always come back to refactor this in the future.